src/overlays/actors/ovl_En_Heishi2/z_en_heishi2.c

```diff
@@ -21,77 +67,880 @@ const ActorInit En_Heishi2_InitVars = {
+        osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆☆☆☆ いやー ついうっかり ☆☆☆☆☆ \n" VT_RST);
+
+        Actor_SetScale(thisx, 0.02f);
+
+        this->unk_274 = thisx->posRot.pos;
+        thisx->posRot.rot.y = 0x7918;
+        thisx->posRot.pos.x += 90.0f;
+        thisx->posRot.pos.y -= 60.0f;
+        thisx->posRot.pos.z += 90.0f;
+        thisx->shape.rot.y = thisx->posRot.rot.y;
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+        func_8002DF54(globalCtx, 0, 8);
+        thisx->flags |= 0x11;
+        this->actionFunc = func_80A544AC;
+        return;
+    }
+    this->unk_2E0 = 60.0f;
+    ActorShape_Init(&thisx->shape, 0.0f, ActorShadow_DrawFunc_Circle, 30.0f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_0600BAC8, &D_06005C30, this->limbDrawTable,
+                   this->transitionDrawTable, 17);
+    collider = &this->collider;
+    Collider_InitCylinder(globalCtx, collider);
+    Collider_SetCylinder(globalCtx, collider, thisx, &cylinderInit);
+    this->collider.dim.yShift = 0;
+    this->collider.dim.radius = 0xF;
+    this->collider.dim.height = 0x46;
+    thisx->unk_1F = 6;
+
+    switch (this->initParams) {
+
+        case 2:
+            this->actionFunc = func_80A531E4;
+            thisx->gravity = -1.0f;
+            break;
+        case 5:
+            this->actionFunc = func_80A53908;
+            thisx->gravity = -1.0f;
+            break;
+        case 6:
+            osSyncPrintf("\n\n");
+            // "Peep hole soldier!"
+            osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ 覗き穴奥兵士ふぃ〜 ☆☆☆☆☆ \n" VT_RST);
+            Collider_DestroyCylinder(globalCtx, collider);
+            thisx->flags &= -0xA;
+            this->actionFunc = func_80A531D8;
+            break;
+    }
+
+    this->unk_2F0 = (thisx->params >> 8) & 0xFF;
+    osSyncPrintf("\n\n");
+    // "Soldier Set 2 Completed!"
+    osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ 兵士２セット完了！ ☆☆☆☆☆ %d\n" VT_RST, thisx->params);
+    // "Identification Completed!"
+    osSyncPrintf(VT_FGCOL(YELLOW) " ☆☆☆☆☆ 識別完了！         ☆☆☆☆☆ %d\n" VT_RST, this->initParams);
+    // "Message completed!"
+    osSyncPrintf(VT_FGCOL(PURPLE) " ☆☆☆☆☆ メッセージ完了！   ☆☆☆☆☆ %x\n\n" VT_RST, (thisx->params >> 8) & 0xF);
+}
+
+void EnHeishi2_Destroy(EnHeishi2* this, GlobalContext* globalCtx) {
+    if ((this->collider.dim.radius != 0) || (this->collider.dim.height != 0)) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+}
+
+void func_80A531CC(EnHeishi2* this, GlobalContext* globalCtx) {
+}
+
+void func_80A531D8(EnHeishi2* this, GlobalContext* globalCtx) {
+}
+
+void func_80A531E4(EnHeishi2* this, GlobalContext* globalCtx) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005C30, 1.0f, 0.0f,
+                         (s16)(f32)SkelAnime_GetFrameCount(&D_06005C30.genericHeader), 0, -10.0f);
+    this->actionFunc = func_80A53278;
+}
+
+// CastleGuard_SetTextId
+void func_80A53278(EnHeishi2* this, GlobalContext* globalCtx) {
```

Comment:
feel free to name this, but the system is EnHeishi2 not CastleGuard.
On second thought, it does a bit more than just set text, including setting the action function